src/components/VersionChooser/index.tsx

```diff
@@ -73,13 +113,16 @@ export class VersionChooserBase extends React.Component<Props> {
 
 const mapStateToProps = (
   state: ApplicationState,
-  ownProps: PublicProps,
+  ownProps: RouterProps,
 ): PropsFromState => {
-  const versionsMap = state.versions.byAddonId[ownProps.addonId];
+  const { byAddonId } = state.versions;
+  const { addonId } = ownProps.match.params;
 
   return {
-    versionsMap,
+    versionsMap: byAddonId[parseInt(addonId, 10)],
   };
 };
 
-export default connect(mapStateToProps)(VersionChooserBase);
+export default withRouter<Partial<DefaultProps> & RouterProps>(
```

Comment:
I would feel better if this started with `PublicProps` (which would be empty) just because we will most definitely forget to add `PublicProps` to this type in case we need to add some later.

Reply:
Oh right!